packages/core/src/QueryBuilder/QueryBuilder.tsx

```diff
@@ -52,7 +54,7 @@ export interface HvQueryBuilderProps {
   /** Max depth of nested query groups. */
   maxDepth?: number;
   /** Object containing all the labels. */
-  labels?: HvQueryBuilderLabels;
+  labels?: DeepPartial<HvQueryBuilderLabels>;
```

Comment:
Note to self for v6: flatten these `labels` so we (and our users) don't have to deal with the nesting complexity
The `rule.value` should probably be replaced with a more streamlined way to configure or override "renderers"  too

Reply:
True, I'll add a comment to not forget this for the next major